services/actions/run.go

```diff
@@ -9,12 +9,76 @@ import (
 
 	actions_model "code.gitea.io/gitea/models/actions"
 	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/log"
 	"code.gitea.io/gitea/modules/util"
+	notify_service "code.gitea.io/gitea/services/notify"
 
 	"github.com/nektos/act/pkg/jobparser"
 	"gopkg.in/yaml.v3"
 )
 
+// PrepareRun prepares a run before inserting it into the database
```

Comment:
```suggestion
// PrepareRun prepares a run and inserts it into the database
```
Before, it read to me like `the caller should call InsertRun themselves`, and I was confused where the `InsertRun` call is…

Reply:
`PrepareRunAndInsert`

Reply:
I will rename this soon, edit renamed in https://github.com/go-gitea/gitea/pull/35624/commits/0cc6d31fdb02052d8b98d82c2ec692e8b4265296